internal/cmd/commands/credentialscmd/userpassword_funcs.go

```diff
@@ -0,0 +1,97 @@
+package credentialscmd
+
+import (
+	"github.com/hashicorp/boundary/api/credentials"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+)
+
+func init() {
+	extraUserpasswordFlagsFunc = extraUserPasswordFlagsFuncImpl
+	extraUserpasswordActionsFlagsMapFunc = extraUserPasswordActionsFlagsMapFuncImpl
+	extraUserpasswordFlagsHandlingFunc = extraUserPasswordFlagHandlingFuncImpl
+}
+
+const (
+	usernameFlagName = "username"
+	passwordFlagName = "password"
+)
+
+type extraUserpasswordCmdVars struct {
+	flagUsername string
+	flagPassword string
+}
+
+func extraUserPasswordActionsFlagsMapFuncImpl() map[string][]string {
+	flags := map[string][]string{
+		"create": {
+			usernameFlagName,
+			passwordFlagName,
+		},
+	}
+	flags["update"] = flags["create"]
+	return flags
+}
+
+func extraUserPasswordFlagsFuncImpl(c *UserpasswordCommand, set *base.FlagSets, _ *base.FlagSet) {
+	f := set.NewFlagSet("User Password Credential Options")
+
+	for _, name := range flagsUserpasswordMap[c.Func] {
+		switch name {
+		case usernameFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   usernameFlagName,
```

Comment:
This strikes me as inconsistent. If we are using `userpassword` instead of `usernamepassword` then the field should be `user`. If we think that it should be `username` -- instead of the more inclusive `user` that might also stand for `userid` or so -- then probably the naming of resources should be `UsernamePassword`, not `UserPassword`.